tools/integration/test/integration/e2e-test-service/definitionTest.js

```diff
@@ -14,19 +14,16 @@ describe('Validation definitions between dev and prod', function () {
   //Rest a bit to avoid overloading the servers
   afterEach(() => new Promise(resolve => setTimeout(resolve, definition.timeout / 2)))
 
-  describe('Validation between dev and prod', function () {
-    before(() => {
-      loadFixtures().forEach(([url, definition]) =>
```

Comment:
Could you please elaborate the reason why loadFixture is removed?  Some definitions are different between Dev and Prod for the following reasons:
- Fixes implemented in Dev, but component is not reharvested on Prod.  Eventually, the component is eventually correct on Prod, then fixture can be removed.
- Feature implemented in Dev but not yet deployed to Prod, e.g. LicenseRef.

Reply:
I've removed the fixtures because their purpose was not clear to me. It's not obvious to me what kind of assurance we expect to get from a test that compares the real (dev) definition to a modified (mocked prod) definition. 

If we expect the definitions to be different, due to a bug fix or a new feature, let's assert that in a test, for each definition separately. Mocking and expecting equality, as is done now, seems to work in the opposite direction, glossing over the expected difference.

I admit I might be misunderstanding the purpose of this whole test, so maybe I should leave it as is and create a new one that will do the comparison using the new comparison function? However my initial approach was to modify this test, and I've removed the mocks to make sure I compare the real data.

Reply:
The goal of integration tests is to detect any breaking changes. For example, after the recent ScanCode upgrade was completed, the integration tests showed differences: some were improvements, and some were regressions (See https://github.com/clearlydefined/service/pull/1056#issuecomment-2176681117). For the improvements that are not in the Production deployment, vetted improved definition files can be put into fixtures, allowing for a successful run before the Production deployment is updated. Ideally, after the production deployment is updated, the components whose definitions can be fixed by the new deployment can be re-harvested and the fixture can be removed, as the definitions in dev and prod should be in sync again.  @elrayle, Feel free to add if anything is missed here.

Reply:
Thanks for the information @qtomlinson. I've decided to keep this test unchanged and add my structured diff comparison as a separate step.

Reply:
The approach looks good.  Another idea just occurred to me: the search query in the getRecentDefinition is based on _meta.updated, which is also the change publication based off.  The changed coordinates published hourly can potentially be used to provide the recent coordinates by day and by type (through sorting).  Just thought to mention it as an idea.

Reply:
It sounds like this is where you already landed.  I'll add my thoughts for clarity.  I like having both, fixture based tests and dynamic tests.  Fixed catches regressions where we expect the same results every time.  Dynamic provides a broader sweep with a goal of increasing confidence or identify systemic problems with a proposed release.